test/groovy/com/sap/piper/FileUtilsTest.groovy

```diff
@@ -11,34 +13,36 @@ import com.sap.piper.FileUtils
 
 class FileUtilsTest {
 
-    @Rule
-    public ExpectedException thrown = new ExpectedException().none()
+    @ClassRule
+    public static TemporaryFolder tmp = new TemporaryFolder()
 
     @Rule
-    public TemporaryFolder tmp = new TemporaryFolder()
+    public ExpectedException thrown = new ExpectedException().none()
 
-    private File emptyDir
-    private File notEmptyDir
-    private File notDir
+    private static emptyDir
+    private static notEmptyDir
+    private static file
 
 
-    @Before
-    void setUp() {
+    @BeforeClass
+    static void createTestFiles() {
 
-        emptyDir = tmp.newFolder('emptyDir')
-        notEmptyDir = tmp.newFolder('notEmptyDir')
-        File file = new File("${notEmptyDir.getAbsolutePath()}${File.separator}test.txt")
-        file.createNewFile()
-        notDir = tmp.newFile('noDir')
+        tmp.newFolder('emptyDir')
+        tmp.newFolder('notEmptyDir')
+        emptyDir = "${tmp.getRoot()}${File.separator}emptyDir"
+        notEmptyDir = "${tmp.getRoot()}${File.separator}notEmptyDir"
+        tmp.newFile("notEmptyDir${File.separator}file")
```

Comment:
`notEmptyDir` should be `${notEmptyDir}`

Reply:
In this case not, because it would create inside the temp dir, and inside the 'notEmptyDir' dir the file 'file'.

Reply:
What about:
```
    emptyDir = tmp.newFolder('emptyDir').getAbsolutePath()
    notEmptyDir = tmp.newFolder('notEmptyDir').getAbsolutePath()
    file = tmp.newFile("notEmptyDir${File.separator}file").getAbsolutePath()
```
it is shorter (less redundant) and less confusing. The string 'notEmptyDir' is still redundant, hence there is still the space for further improvement :-)

Reply:
Ok. The second sentence I don't get it. Don't you like the name 'notEmptyDir' or what do you mean?

Reply:
No, no, it is not about liking / dis-liking the name. All I say is: we have the string 'notEmptyDir' twice in the code snippet above, denoting in both cases the same thing.
An "orthodox" developer will most probably fight against such redundancies and asking for doing is like this

```
def notEmptyDirName = 'notEmptyDir'
emptyDir = tmp.newFolder('emptyDir').getAbsolutePath()
notEmptyDir = tmp.newFolder(notEmptyDirName).getAbsolutePath()
file = tmp.newFile("${notEmptyDirName}${File.separator}file").getAbsolutePath()
```

:-) --> ;-) --> :-)